### PR TITLE
[SPARK-22916][SQL][FOLLOW-UP] Update the Description of Join Selection

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
This PR is to update the description of the join algorithm changes.

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @liufengdb @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala [hunk 1]
```diff
@@ -91,23 +91,58 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
...
+   *     1) Broadcast the join side with the broadcast hint, even if the size is larger than
+   *     [[SQLConf.AUTO_BROADCASTJOIN_THRESHOLD]]. If both sides have the hint (i.e., just for
+   *     inner-like join), the side with a smaller estimated physical size will be broadcast.
+   *     2) Respect the [[SQLConf.AUTO_BROADCASTJOIN_THRESHOLD]] threshold and broadcast the side
+   *     whose estimated physical size is smaller than the threshold. If both sides are below the
+   *     threshold, broadcast the smaller side. If neither is smaller, BNLJ is not used.
+   *
+   * - CartesianProduct: for inner like join, CartesianProduct is the fallback option.
+   *
+   * - BroadcastNestedLoopJoin (BNLJ):
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala]: `BroadcastNestedLoopJoin (BNLJ)` appeared twice in this list, is it intentional?

**Comment by cloud-fan** [comment]
maybe we should separate the distribution concern and algorithm concern from the join selection when documenting it:
* for distribution concern, we can do broadcast, shuffle or cartesian
* for algorithm concern, we can do hash join, sort merge join or nested loop join.

By combining these 2 concerns, we get many different join selections(exclude some invalid combinations):
* broadcast hash join
* broadcast nested loop join
* (shuffled) sort merge join
* shuffled hash join
* cartesian product (nested loop join)

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala (hunk 1 above)
Comment (gatorsmile) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala]: This is based on the order of our JoinSelection rule

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
`org.apache.spark.sql.execution.datasources.orc.OrcQuerySuite.(It is not a test it is a sbt.testing.SuiteSelector)` is another flaky test

**Comment by cloud-fan** [comment]
retest this please

**Comment by liufengdb** [comment]
LGTM! Thanks for doing this!

**Comment by gatorsmile** [comment]
Thanks! Merged to master/2.3. 

This PR is just to reflect the latest changes in the join selection rule. We can continue the improvement in the comment.

**Pull request closed without merging**